lib/puppet/type/file/data_sync.rb

```diff
@@ -27,7 +27,8 @@ def checksum_insync?(param, is, has_contents, &block)
         return false if is == :absent
       else
         if resource[:ensure] == :present && has_contents && (s = resource.stat)
-          resource.warning "Ensure set to :present but file type is #{s.ftype} so no content will be synced"
+          #TRANSLATORS ':present' is a Puppet attribute value and should not be translated
+          resource.warning _("Ensure set to :present but file type is %{file_type} so no content will be synced") % { file_type: s.ftype}
```

Comment:
Isn't `ensure` also an attribute name that we don't want translated?

Reply:
yes you are correct